fri/src/folder.rs

```diff
@@ -15,7 +15,7 @@ impl MultiplicativeFriFolder {
         Self::fold(f_x, f_minus_x, eval_point, &x_inv)
```

Comment:
Calculate x_inv once and reuse over-and-over, currently computing this one-off inversion should dominate the folding cost.

Reply:
Consider batch inverting all the x's across the FRI layers as well.